README.md

```diff
@@ -142,39 +99,110 @@ execute one of the following make commands. This step is not necessary in case t
 |  `make unknown`         |  Compiles the source-code for architectures other than X86_64, e.g. RISC-V, ARM, etc. |
 |  `make clean`           |  Cleaning-up all binary files.                                                        |
 
+# API - Code example for a simple benchmark
+
+We now describe a very simple example-benchmark that uses the Application Programming Interface (API) of the provided runtime. This simple benchmark measures the performance of Fetch&Add instructions in multi-core machines.
+
+```c
+#include <stdio.h>
+#include <stdint.h>
+
+#include <primitives.h>
+#include <threadtools.h>
+#include <barrier.h>
+
+#define N_THREADS 10
+#define RUNS      1000000
+
+volatile int64_t object CACHE_ALIGN;
+int64_t d1 CACHE_ALIGN, d2;
+Barrier bar CACHE_ALIGN;
+
+inline static void *Execute(void *Arg) {
+    long i, id = (long)Arg;
+
+    BarrierWait(&bar);
+    if (id == 0) d1 = getTimeMillis();
+
+    for (i = 0; i < RUNS; i++)
+        FAA64(&object, 1);
+
+    BarrierWait(&bar);
+    if (id == 0) d2 = getTimeMillis();
+
+    return NULL;
+}
+
+int main(int argc, char *argv[]) {
+    object = 1;
+
+    BarrierSet(&bar, N_THREADS);
+    StartThreadsN(N_THREADS, Execute, _DONT_USE_UTHREADS_);
+    JoinThreadsN(N_THREADS - 1);
+
+    printf("time: %ld (ms)\tthroughput: %.2f (millions ops/sec)\n", 
+           (d2 - d1), RUNS * N_THREADS / (1000.0 * (d2 - d1)));
+
+    return 0;
+}
+```
+
+This example-benchmark creates `N_THREADS`, where each of them executes `RUNS` Fetch&Add operations in a shared 64-bit integer. At the end of the benchmark the throughput (i.e. Fetch&Add operations per second) is calculated.
+
+The `StartThreadsN` function (provided by the API defined in `threadtools.h`) in main, creates `N_THREADS` threads and each of the executes the `Execute` function declared in the same file. The `_DONT_USE_UTHREADS_` argument imposes `StartThreadsN` to create only Posix threads; in case that the user sets this argument to some `M` > 0, then `StartThreadsN` will create `N_THREADS` Posix threads and each of them will create `M` user-level (i.e. fiber) threads. The `JoinThreadsN` function (also provided by `threadtools. h`) waits until all Posix and fiber (if any) threads finish the execution of the `Execute` function. The Fetch&Add instruction on 64-bit integers is performed by the `FAA64` function provided by the API of `primitives.h`.
```

Comment:
Maybe `in case that the user sets this argument to` --> `in case that the user sets the corresponding fibers argument to`

Reply:
Fixed.